### PR TITLE
Validate hook names in kontena.yml

### PR DESCRIPTION
**Pull request by nevalla**

### CODE REVIEW

File: cli/lib/kontena/cli/stacks/yaml/custom_validators/hooks_validator.rb [hunk 1]
```diff
@@ -6,10 +6,16 @@ def initialize
 
     def validate(key, value, validations, errors)
       unless value.is_a?(Hash)
-        errors[key] = 'hooks must be array'
+        errors[key] = 'hooks must be hash'
```
Comment (kke) [inline cli/lib/kontena/cli/stacks/yaml/custom_validators/hooks_validator.rb]: I think the yaml terminology is "mapping" = hash, "list" = array

Comment (nevalla) [inline cli/lib/kontena/cli/stacks/yaml/custom_validators/hooks_validator.rb]: Just looked what is used elsewhere. But agree on that.

Comment (nevalla) [inline cli/lib/kontena/cli/stacks/yaml/custom_validators/hooks_validator.rb]: I suggest that we will replace all in once in separate PR.

Comment (kke) [inline cli/lib/kontena/cli/stacks/yaml/custom_validators/hooks_validator.rb]: The volume PR has several uses of "mapping". 

```ruby
lib/kontena/cli/stacks/yaml/custom_validators/build_validator.rb
9:        errors[key] = 'build must be string or hash'

lib/kontena/cli/stacks/yaml/custom_validators/extends_validator.rb
9:        errors[key] = 'extends must be string or hash'
```

I think these are the only ones talking about "hash". I don't know what HashValidator says on a type error, probably not mapping.

Comment (kke) [inline cli/lib/kontena/cli/stacks/yaml/custom_validators/hooks_validator.rb]: also most of the errors seem to say something like `must be string` , should be `must be a string`.

Comment (nevalla) [inline cli/lib/kontena/cli/stacks/yaml/custom_validators/hooks_validator.rb]: I have updated error message to use 'mapping' instead of 'hash'

**Pull request merged**